xarray/tests/test_concat.py

```diff
@@ -978,6 +979,34 @@ def test_concat_str_dtype(self, dtype, dim) -> None:
 
         assert np.issubdtype(actual.x2.dtype, dtype)
 
+    def test_concat_avoids_index_auto_creation(self) -> None:
```

Comment:
Reminder to myself: The reason this test didn't catch the problem described in https://github.com/TomNicholas/VirtualiZarr/issues/18#issuecomment-2023955860 is because this test checks that concatenating datasets that start without indexes stay without indexes, whereas that problem is from concatenating datasets with indexes but having the coordinate variables be silently replaced by `IndexVariable` objects created from the index data.